### PR TITLE
Branch levels

### PR DESCRIPTION
**Pull request by krl**
Expose the `levels` of a Branch for inspection.

### CODE REVIEW

**Comment by krl** [comment]
fixes #46

File: src/walk.rs [hunk 1]
```diff
@@ -77,3 +81,43 @@ where
         unreachable!()
     }
 }
+
+/// Trait that provides a nth() method to any Compound with a Cardinality
+/// annotation
```
Comment (ZER0) [inline src/walk.rs:86]: I believe the comments are copied from the nth, this should said `a first() method`, is that correct?

File: src/walk.rs [hunk 2]
```diff
@@ -77,3 +81,43 @@ where
...
+pub trait First<'a, A>
+where
+    Self: Compound<A>,
+    A: Combine<Self, A>,
+{
+    /// Construct a `Branch` pointing to the `nth` element, if any
```
Comment (ZER0) [inline src/walk.rs:92]: here as well

File: src/walk.rs [hunk 3]
```diff
@@ -77,3 +81,43 @@ where
...
+    fn first(&'a self) -> Result<Option<Branch<'a, Self, A>>, CanonError>;
+
+    /// Construct a `BranchMut` pointing to the `nth` element, if any
```
Comment (ZER0) [inline src/walk.rs:95]: And probably here too.

**Review by ZER0 — CHANGES_REQUESTED** [review]
LGTM, with the comments fixed.

**Review by ZER0 — APPROVED** [review]

**Pull request merged**